feat: Initial Integration of Go Corset

DavePearce opened the pull request:
This updates the github action to run a separate job which uses `go-corset` alongside Rust `corset`.

DavePearce committed:
Initial Integration of `go-corset`

This integrates `go-corset` and (when installed) runs it side-by-side
the existing Rust tool.  However, if the `go-corset` binary cannot be
found then it is simply ignored.

At this stage, `go-corset` is not enabled in the CI pipeline.  This is
because it is consuming too much memory on the test `leoFailingRange`.

powerslider commented on arithmetization/src/main/java/net/consensys/linea/corset/RustCorsetValidator.java:
```diff
@@ -0,0 +1,228 @@
+   * @return
+   */
+  private List<String> buildCommandLine(Path traceFile, String zkEvmBin) {
+    ArrayList<String> options = new ArrayList<>();
```
Please use the interface type here `List<String>`.

DavePearce replied:
Ok, done.  Also rebased to HEAD for `arith-dev`.

DavePearce committed:
Refactor ArrayList<String> => List<String>

Minor refactoring of method variable to use an interface type instead of
a concrete class.

powerslider committed:
Merge branch 'arith-dev' into feat/issue-906/initial-integration-of-go-corset

powerslider committed:
Merge branch 'arith-dev' into feat/issue-906/initial-integration-of-go-corset
